web/lib/components/flashcards/PageAllSets.jsx

```diff
@@ -50,22 +81,60 @@ export function PageAllSets({ page }) {
     )
   }
 
+  const doCreateNewSet = (name) => async (e) => {
+    e.preventDefault()
+
+    await createNewSet.mutateAsync({ name, userId: user.id })
+
+    if (createNewSet.status !== 'error') {
+      notify({ title: _e('flashcardset.createNewSetSuccessfully') })
+    } else {
+      notify({
+        title: _e('flashcardset.error.createNewSet'),
+        variant: 'error',
+      })
+    }
+
+    setIsDialogOpen(false)
+  }
+
   if (isLoading) return <Layout loading />
 
   const links = [
     { href: '/', name: _e('nav.home') },
     { href: '/flashcards', name: _e('nav.flashcards') },
   ]
 
+  const openDialog = () => setIsDialogOpen(true)
+
   return (
     <Layout>
       <div className="container">
-        <div className="mb-4">
-          <Breadcrumb links={links} />
+        <div className="mb-2 grid w-full grid-cols-2 grid-rows-1">
+          <div className="flex w-full items-center">
+            <Breadcrumb links={links} />
+          </div>
+          <div className="flex w-full flex-row items-center justify-end">
+            <Button
+              className="grid-rows-2-min-fr grid auto-cols-min grid-cols-2 gap-2 rounded border border-sky-300 bg-sky-100 p-2 text-sm font-semibold text-sky-700 hover:border-sky-400"
+              onClick={openDialog}>
+              <PlusIcon className="h-5 w-5" />{' '}
+              <span className="">{_e('flashcardset.create')}</span>
+            </Button>
+          </div>
         </div>
+        <CreateNewSetDialog
+          onCreateNewSet={doCreateNewSet}
+          open={isDialogOpen}
+          onOpenChange={setIsDialogOpen}
+        />
 
         <div className="mb-4">
-          <Sets flashcardSets={flashcardSets} />
+          <Sets
+            flashcardSets={flashcardSets}
+            updateSet={updateSet}
+            deleteSet={deleteSet}
```

Comment:
Lift it up, change to `onSetUpdate` and `onSetDelete` then handle in parent component.